src/Sylius/Bundle/CoreBundle/EventListener/OrderPaymentListener.php

```diff
@@ -71,6 +71,8 @@ public function createOrderPayment(GenericEvent $event)
     {
         if (false === $this->getOrder($event)->getLastPayment()) {
             $this->paymentProcessor->createPayment($this->getOrder($event));
+        } else {
+            $this->getOrder($event)->getLastPayment()->setDetails(array());
         }
```

Comment:
I do not think we have to change the last payment but rather create a new one so an admin can see the whole history, like user selected one payment but than changed his mind and selected another one